tiledb/sm/filter/filter.h

```diff
@@ -85,6 +86,34 @@ class Filter {
    */
   Status get_option(FilterOption option, void* value) const;
 
+  /**
+   * @brief Returns the filter output type
+   *
+   * @param input_type Expected type used for input. Used for filters which
+   * change output type based on input data. e.g. XORFilter output type is
+   * based on byte width of input type.
+   */
+  virtual Datatype output_datatype(Datatype input_type = Datatype::ANY) const;
+
+  /**
+   * @brief Throws if given filter's output *cannot* be handled by this filter.
+   *
+   */
+  virtual void ensure_compatible_output(const Filter& filter) const;
+
+  /**
+   * @brief Throws if given data type *cannot* be handled by this filter.
+   *
+   */
+  virtual void ensure_accepts_datatype(Datatype datatype) const;
+
+  /**
+   * @brief Checks if the filter is applicable to the input datatype.
+   *
+   * @param type Input datatype to check filter compatibility.
+   */
+  virtual bool accepts_input_datatype(Datatype type) const;
```

Comment:
Is it just me or do all three of these functions basically do the same thing but differ only on their input and whether they throw vs not?

Reply:
You're correct, and looking again `ensure_compatible_output(Filter)` is unused, so for now I'll drop that function and keep the other two. I thought the boolean variant could be useful in situations where throwing isn't required but I'm also happy to combine into just `ensure_accepts_datatype`.